microsoft/testsuites/core/msr.py

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from assertpy import assert_that
+
+from lisa import (
+    RemoteNode,
+    SkippedException,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.operating_system import CBLMariner, Debian, Fedora, Linux, Suse
+from lisa.tools import Modprobe
+
+# See docs for hypercall spec, sharing os info
+# is required before making hypercalls.
+#  https://learn.microsoft.com/en-us/virtualization/hyper-v-on-windows/tlfs/hypercall-interface#guest-os-identity-for-proprietary-operating-systems
+# build info: bits 0-15
+BUILD_INFO = 0x0000_0000_0000_FFFF
+# service pack, bits 16-23
+SERVICE_PACK = 0x0000_0000_00FF_0000
+# minor version, bits 24-31
+MINOR_VERSION = 0x0000_0000_FF00_0000
+# major version, bits 32-39
+MAJOR_VERSION = 0x0000_00FF_0000_0000
+# OS ID  bits 40-47
+OS_ID = 0x0000_FF00_0000_0000
+# vendor ID: bits 48-62
+VENDOR_ID = 0x7FFF_0000_0000_0000
+# the all important 'is open source os' flag.
+# indicates whether is *nix or other non-winodws
+# os.  the final bit, bit 63
+IS_OPEN_SOURCE_OS = 0x8000_0000_0000_0000
+
+
+@TestSuiteMetadata(
+    area="core",
+    category="functional",
+    description="""
+    Test suite verifies hyper-v platform id is set correctly via hypercall to host.
+    """,
+    requirement=simple_requirement(supported_os=[Linux]),
```

Comment:
should we use supported platform in case it fails in other platforms?

Reply:
Updated